### DISCUSSION TITLE
модули для ненаучной работы предполагаются?

### DISCUSSION THREAD

**Discussion by Apollon-Theos** [pascalabcnet/pascalabcnet#2826] [Ideas]
Например:
LDAP
SSH
HTTP(S)/FTP(S)
WinRM

а работа с базами данных?
MS SQL
PostgeSQL
MySQL (MariaDB)
MS Access
MS Excel
LibreOffice Calc
LibreOffice Base

или всё же придётся оставаться на php?

**Comment by Apollon-Theos** [comment]
можно добавить в списоу файлы .csv, с автоматическим импортом в массив или коллесцию, и экспортом в .csv

**Comment by samuraiGH** [comment]
А зачем, если есть nuget?

**Comment by Apollon-Theos** [comment]
а зачем Pascal, если есть Assembler?
а зачем PascalABC если есть MS Excel (LibreOffice Calc)?
а зачем учить PascalABC.Net если его после экзаменов негде применять? Учебник "История КПСС" очень хорошая иллюстрация к такому подходу.

Вспоминая BP7 и его TurboVision, не стоит забывать и о TECHNOJOCK'S OBJECT TOOLKIT, в котором TUI рисовался лучше.

Наличие стандартных, проверенных разработчиками, модулей, входящих в комплект поставки, характеризует зрелость продукта. А не заточку на обучение программированию, с постоянными доработками с надёрганными из других языков программирования фичами. Пусть эти модули не будут супер оптимизированными, но будут. 
Иначе всё и все вернутся к MS Excel, на котором умельцы делают замену 1С, почти полную. А уж расчёты там и графики, прям из коробки :)

**Reply by samuraiGH** [reply]
приведите пример таких модулей в других .net языках

**Reply by samuraiGH** [reply]
зачем пахать поле, если его уже вспахали?...

**Reply by Apollon-Theos** [reply]
> зачем пахать поле, если его уже вспахали?...

а чем занимаются создатели PascalABC.Net, если не "повторным вспахиванием" ? FreePascal всё еще жив и работает на большем количестве платформ. Причем исходники Delphi с минимальными изменениями перевомпилирует :)

**Reply by Apollon-Theos** [reply]
> приведите пример таких модулей в других .net языках

а я про что писал? "надергивание фич из других языков програмирования"
Кстати очень стрнная бурная реакция, не находите? Может действительно больной вопрос? Тот же php не стесняется включать встандартную бесплатную поставку максимальное количество модулей, сопрягающих с множеством иныраструктурных внедрённых решений.

**Reply by samuraiGH** [reply]
> а чем занимаются создатели PascalABC.Net, если не "повторным вспахиванием" ?

Вы злоупотребляете "риторикой". Пусть будет по вашему (хотя это не так) и разработчики перепахивают поле компиляторов. Зачем им перепахивать ещё и поле библиотек, если и в компиляторе работы достаточно?
> Может действительно больной вопрос?

Может быть всё-таки приведёте пример, что бы мы не обсуждали абстрактные модули? Пока вы только предложили переписать всё пространство имён System.Net, библиотеки Npgsql, Microsoft.Office.Interop.Excel, etc. Вот и хотелось бы узнать, кто ещё из .net языков решил пойти в разрез .net и внедрять свои велосипеды

**Reply by Apollon-Theos** [reply]
Вот здесь не понял, "свои велосипеды" это когда потом в интернете ищешь, что же не работает, твоя прога или какой-то из nuget-package? вспомним log4j ?
Видимо я был не понят, никто не требует отказа от всего множества nupkg, но например у нас на работе их запрещено скачивать из интернета. Есть проверенный набор в проверенных версиях и крутись как можешь. В этих реалиях, как раз и наличие в стендартной поставке минимального работа модулей и интерфейсов к ним, было бы очень полезно.

**Reply by Apollon-Theos** [reply]
И наличие в nuget порядка 40 package для работы с Active Directory, тоже можно считать недостатком. Который из них работает? А правильно работает? А все ли функции там есть? 
Если разработчик включает в стандартную поставку какой-то модуль, то он гарантирует положительные ответы на все эти вопросы.

А сидеть перебирать 40 можудей, для того чтобы найти тот работающий, никакой работодатель не даст. Да и комп потом чистить надо будет от остатков "предыдущий внедрений".

**Reply by SunSerega** [reply]
> вспомним log4j ?

А если и так перегруженные разработчики, работающие в своё свободное время, сделают свой велосипед - вы думаете вероятность таких проблем будет меньше?

В конце концов, исходный код открытый и предложения модулей в виде пулов рассматривают.
Но я считаю что это заведомо неправильный подход. Вы пытаете исправить проблему менеджмента на вашей работе, выдвигая претензии к устоявшейся системе из вне.

> И наличие в nuget порядка 40 package для работы с Active Directory, тоже можно считать недостатком. Который из них работает? А правильно работает? А все ли функции там есть?

Богатство вариантов это никогда не недостаток. Учитесь сортировать информацию, использовать агрегаторы и просто тестировать.
Я понимаю что когда проприетарная система за вас всё заранее разжёвывает это приятно. Но массы делающие модули/библиотеки всегда могут добиться бОльшего. И в бесплатном проекте как тут - это единственный вариант.

**Reply by Apollon-Theos** [reply]
Я не пытаюсь требовать от разработчиков каких-либо обязательств, я задал вопрос, в самом начале. То что они делают - это их выбор, и препятствоватьим в этом тоже не стоит, как и восхвалять.
Жаль, что enterprise решения сначала считают "ошибками менеджмента", и жаль что это разработчики. Я надеюсь change management Вам всё же знакомо. И поверьте, как сопровождающий специалист, я полной чашей хлебнул его нарушения, как раз со стороны разработчиков. В том числе и на production контуре.
Оттуда же и полное нежелание тратить время на "богатство вариантов", возвращаясь к тематике велосипедов, "на тебе раму, а вон там свалка колёс, насобирай себе что-нибудь, вдруг поехать получится".
Надеюсь я правильно понял Вашу политику :) И очень часто модули в open source делают очень урезанными, для решения узкой задачи "здесь и сейчас", а потом выкладывают, типа "не жалко". :) Тоже проходилось не раз, потому и задал свой вопрос.

**Comment by Apollon-Theos** [comment]
а nuget - это уже если не хватит возможностей стандартных модулей из поставки.

**Comment by Apollon-Theos** [comment]
а вдруг TOT удасться скомпилировать под PascalABC.Net ? TUI иногда полезнее GUI :)

**Comment by Apollon-Theos** [comment]
Я не спорю, что среда обучения нужна, очень хорошо, что она частично российская, но оставаться на тех же позициях, что и GWBasic, примитивно. 
Вполне возможно появлении версии "Pro" с теми самыми модулями для работы в Enterprise среде, особенно если цена будет очень небольшая, то распространение ее может и помочь проекту.
PowerShell с его рекламируемой всеядностью, но убогой обработкой ошибок и зависимостью от скриптового наследия, может быть подвинут PascalABC.Net если в нём будут возможности о которых тут дискутируем.

**Reply by samuraiGH** [reply]
Это не среда обучения, а язык общего назначения

**Comment by ibond84** [comment]
Поставил вот хваленый Lazarus в Ubuntu, а он не запускается..

**Reply by Apollon-Theos** [reply]
Lazarus у меня и под виндой не запустился, я про него молчу. FPC реально работает и почти все мои наработки со времен BP 7 перекомпилирует. Из тех что я пробовал, ради проверки совместимости. Даже прожевал и выплюнуо работоспособный вариант из BP for Windows :)

**Comment by Apollon-Theos** [comment]
кстати - это еще один минус PascalABC.Net - это остуствие редактора форм или хотя бы качественного импорта из Delphi.
"Рисовать" формы как есть сейчас с бесконечным ручным редактированием позиции и размера каждого элемента в виде кода инициализации, очень утомительно.
Видимо отсюда и попытка свести всё к рисование на канве основного окна, или использование его в качестве текстового эмулятора

**Reply by samuraiGH** [reply]
Я не совсем понял. Вы редактор форм pabc вообще не видели, или он Вам не понравился?

**Reply by Apollon-Theos** [reply]
вот сижу и пытаюсь найти эту штуку в стандартной поставке, не вижу
на сайте отдельно скачать, не вижу
внутри IDE в меню, не вижу

выходит не видел :(

**Reply by Apollon-Theos** [reply]
**Приношу свои извинения**, здесь я точно большей часью был не прав. Нашёл и изучаю.
Результаты потом напишу, если будет желание со стороны разработчиков.

**Reply by Apollon-Theos** [reply]
А среди компонентов не нашёл Grid, ну чтоб было можно достатояно просто нарисовать шахматную доску :) DataGrid есть, просто Grid возможно спрятан, как и ImageDrid. Придётся по старинке на Panel рисовать :)

**Comment by Apollon-Theos** [comment]
предлагаю закрыть дискуссиию, так как стороны высказались
